### PR TITLE
Client plugin

### PR DESCRIPTION
**Pull request by loosebazooka**

### CODE REVIEW

**Comment by loosebazooka** [comment]
@tangiel @patflynn @meltsufin 
Review #1 first

The second half of the endpoints plugin is the client side which doesn't explicitly define any tasks (couple of hidden tasks that do all the work), it exposes an extension

```
endpointsClient {
  discoveryDocs = [file1, file2, dir2]
}
```

and a dependency configuration "endpointsServer"

```
dependencies {
  endpointsServer project(path ":server", configuration: "discovery-docs")
}
```

and will generate and add source to the project depending on what was specified.

The user will be expected to add the api client dependency to their project on their own.

```
dependencies {
  compile "com.google.api-client:google-api-client:+"
}
```

File: src/main/java/com/google/cloud/tools/gradle/endpoints/client/EndpointsClientPlugin.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Plugin definition for Endpoints Clients (will generate source when applied)
+ *
+ * All tasks are hidden. Configuration of source discovery docs is from two ways:
+ * 1. specify the location of the discovery doc with the extension
+ * endpointsClient {
```
Comment (patflynn) [inline src/main/java/com/google/cloud/tools/gradle/endpoints/client/EndpointsClientPlugin.java:41]: {@code..}

File: src/main/java/com/google/cloud/tools/gradle/endpoints/client/EndpointsClientPlugin.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Plugin definition for Endpoints Clients (will generate source when applied)
+ *
+ * All tasks are hidden. Configuration of source discovery docs is from two ways:
```
Comment (patflynn) [inline src/main/java/com/google/cloud/tools/gradle/endpoints/client/EndpointsClientPlugin.java:39]: could there be use-cases where people might want a bit more control about how this works?  I like the automated nature of this, but I'm wondering if it's flexible enough.

**Review by patflynn — COMMENTED** [review]
Are these PRs effectively based on their parent commits?  I feel like I'm looking at the whole thing each time.

**Comment by loosebazooka** [comment]
closed by #3

**Pull request closed without merging**